include/aws/common/array_list.inl

```diff
@@ -173,6 +173,9 @@ int aws_array_list_push_front(struct aws_array_list *AWS_RESTRICT list, const vo
     if (err_code && aws_last_error() == AWS_ERROR_INVALID_INDEX && !list->alloc) {
         AWS_POSTCONDITION(aws_array_list_is_valid(list));
         return aws_raise_error(AWS_ERROR_LIST_EXCEEDS_MAX_SIZE);
+    } else if (err_code) {
+        AWS_POSTCONDITION(aws_array_list_is_valid(list));
+        return err_code;
```

Comment:
**Suggestion.** If we don't need to return the error signal `AWS_ERROR_LIST_EXCEEDS_MAX_SIZE`, we could just propagate the error from `aws_array_list_ensure_capacity`, which is the actual source of the problem. So, I'd still advise to simplify this function in the following way:
```c
AWS_STATIC_IMPL
int aws_array_list_push_front(struct aws_array_list *AWS_RESTRICT list, const void *val) {
    AWS_PRECONDITION(aws_array_list_is_valid(list));
    AWS_PRECONDITION(
        val && AWS_MEM_IS_READABLE(val, list->item_size),
        "Input pointer [val] must point writable memory of [list->item_size] bytes.");
    size_t orig_len = aws_array_list_length(list);
    int err_code = aws_array_list_ensure_capacity(list, orig_len);
    if (!err_code) {
        if (orig_len) {
            memmove((uint8_t *)list->data + list->item_size, list->data, orig_len * list->item_size);
        }
        ++list->length;
        memcpy(list->data, val, list->item_size);
    }
    AWS_POSTCONDITION(aws_array_list_is_valid(list));
    return err_code;
}
```